server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -338,130 +342,427 @@ describe("Routerlicious", () => {
+											count == 0 ? signalContent1 : signalContent2,
+											signal.content,
+											"Signal content mismatch",
+										);
+										assert.equal(
+											firstConnectMessage.clientId,
+											signal.clientId,
+											"Client ID mismatch",
+										);
+										count++;
+
+										if (count == 2) {
+											resolve();
+										}
+									});
+								}),
+							];
+
+							firstSocket.send("submitSignal", firstConnectMessage.clientId, [
+								signalContent1,
+							]);
+							firstSocket.send("submitSignal", firstConnectMessage.clientId, [
+								signalContent2,
+							]);
+							await Promise.all(promises);
+						});
 
-						const secondSocket = webSocketServer.createConnection();
+						it("Invalid clientID", async () => {
+							const promises = [
+								new Promise<void>((resolve) => {
+									firstSocket.on("signal", (signal: ISignalMessage) => {
+										assert.fail("Signal should not have been received");
+									});
+									setTimeout(resolve, 100);
+								}),
+								new Promise<void>((resolve) => {
+									secondSocket.on("signal", (signal: ISignalMessage) => {
+										assert.fail("Signal should not have been received");
+									});
+									setTimeout(resolve, 100);
+								}),
+							];
+
+							firstSocket.send("submitSignal", "invalidClientID", ["TestSignal"]);
+							await Promise.all(promises);
+						});
+					});
 
-						await connectToServer(testId, testTenantId, testSecret, secondSocket);
-
-						const promises = [
-							new Promise<void>((resolve) => {
-								firstSocket.on("signal", (signal: ISignalMessage) => {
-									assert.equal(
-										signalContent,
-										signal.content,
-										"Signal content mismatch",
-									);
-									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
-									);
-									resolve();
-								});
-							}),
-							new Promise<void>((resolve) => {
-								secondSocket.on("signal", (signal: ISignalMessage) => {
-									assert.equal(
-										signalContent,
-										signal.content,
-										"Signal content mismatch",
-									);
-									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
-									);
-									resolve();
+					describe("v2 signals", () => {
+						let userSignalCounts: number[];
+
+						const createSignalPromise = (
+							userIndex: number,
+							socket: LocalWebSocket,
+						): Promise<ISignalMessage> => {
+							return new Promise((resolve) => {
+								socket.on("signal", (signal: ISignalMessage) => {
+									userSignalCounts[userIndex]++;
+									resolve(signal);
 								});
-							}),
-						];
+							});
+						};
 
-						firstSocket.send("submitSignal", firstConnectMessage.clientId, [
-							signalContent,
-						]);
-						await Promise.all(promises);
-					});
+						let firstSocket: LocalWebSocket;
```

Comment:
One way to organize this better is to create type for each client and create array of those types. Would likely be less duplication, more readable syntax, and ability to extend and run tests across more clients if/when needed.